api/resolver.py

```diff
@@ -180,47 +364,34 @@ class Resolver(object):
 
     Does not tolerate ambiguity at any level of the path except the final node.
     """
+    def __init__(self, id_only=False):
+        self.id_only = id_only 
 
-    @staticmethod
-    def resolve(path):
-
+    def resolve(self, path):
```

Comment:
This function is pretty slick relative to the previous one. I'm a little iffy on the callees modifying the passed variables, but I can see the advantage.